src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -633,6 +636,17 @@ private void CompileNamedType(NamedTypeSymbol containingType)
             compilationState.Free();
         }
 
+        private static ImmutableArray<SyntaxNode> GetSyntaxNodesForConstructorNullableEnabledContext(NamedTypeSymbol containingType)
```

Comment:
>GetSyntaxNodesForConstructorNullableEnabledContext [](start = 50, length = 50)

Do we get here for a simple program entry point type symbol and do we get the right result for it?

Reply:
Yes, we get here for "<Main>$", and `sourceType.SyntaxReferences[0].GetSyntax()` is the entire `CompilationUnit` which seems appropriate.

---
In reply to: [544660569](https://github.com/dotnet/roslyn/pull/49953#discussion_r544660569) [](ancestors = 544660569)

Reply:
> Yes, we get here for "<Main>$", and sourceType.SyntaxReferences[0].GetSyntax() is the entire CompilationUnit which seems appropriate.

I do not think this is appropriate. According to the language specification, the actual body contains only global statements. Usings, types and namespace declarations should be excluded in my opinion. I assume primary constructors have similar issue. For those, only base arguments should be included  and perhaps parameters list, I think




---
In reply to: [544720796](https://github.com/dotnet/roslyn/pull/49953#discussion_r544720796) [](ancestors = 544720796,544660569)

Reply:
It is quite possible that this doesn't make much difference for this helper, which, as I commented elsewhere, in my opinion introduces too much relaxation on its own. However, that is likely to make a difference elsewhere where we actually decide if we need to perform analysis for the body of the method. 

---
In reply to: [544723672](https://github.com/dotnet/roslyn/pull/49953#discussion_r544723672) [](ancestors = 544723672,544720796,544660569)